visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -134,3 +151,70 @@ function getAsApiVersion(version: string): ApiVersion {
 		minor: Number(version.split(".")[1])
 	}
 }
+
+function validateFixedFolders(file: ExportCCFile, result: CCValidationResult) {
+	const notFixed: string[] = []
+	const outOfBounds: string[] = []
+	const intersections: Set<string> = new Set()
+
+	file.nodes[0].children.forEach(node => {
+		if (node.fixed === undefined) {
+			notFixed.push(node.name)
+		} else {
+			if (isOutOfBounds(node)) {
+				outOfBounds.push(node.name)
+			}
+
+			file.nodes[0].children.forEach(node2 => {
+				if (node !== node2 && rectanglesIntersect(node.fixed, node2.fixed)) {
+					if (!intersections.has(`${node2.name} and ${node.name}`)) {
+						intersections.add(`${node.name} and ${node2.name}`)
+					}
+				}
+			})
+		}
+	})
+
+	if (notFixed.length > 0 && notFixed.length !== file.nodes[0].children.length) {
+		result.title = ERROR_MESSAGES.notAllFoldersAreFixed.title
+		result.error.push(`${ERROR_MESSAGES.notAllFoldersAreFixed.message} Found: ${notFixed.join(", ")}`)
+	}
+
+	if (outOfBounds.length > 0) {
+		result.title = ERROR_MESSAGES.fixedFoldersOutOfBounds.title
+		result.error.push(`${ERROR_MESSAGES.fixedFoldersOutOfBounds.message} Found: ${outOfBounds.join(", ")}`)
+	}
+
+	if (intersections.size > 0) {
+		result.title = ERROR_MESSAGES.fixedFoldersOverlapped.title
+		result.error.push(`${ERROR_MESSAGES.fixedFoldersOverlapped.message} Found: ${[...intersections.values()].join(", ")}`)
+	}
+}
+
+function rectanglesIntersect(rect1: Fixed, rect2: Fixed): boolean {
+	if (rect1 !== undefined && rect2 !== undefined) {
+		return (
+			isInRectangle(rect1.x, rect1.y, rect2) ||
+			isInRectangle(rect1.x, rect1.y + rect1.height, rect2) ||
+			isInRectangle(rect1.x + rect1.width, rect1.y, rect2) ||
+			isInRectangle(rect1.x + rect1.width, rect1.y + rect1.height, rect2)
+		)
+	}
+}
+
+function isInRectangle(x: number, y: number, rect: Fixed): boolean {
+	return x >= rect.x && x <= rect.x + rect.width && y >= rect.y && y <= rect.y + rect.height
+}
+
+function isOutOfBounds(node: CodeMapNode): boolean {
+	return (
+		outOfRange(node.fixed.x) ||
+		outOfRange(node.fixed.y) ||
+		outOfRange(node.fixed.x + node.fixed.width) ||
+		outOfRange(node.fixed.y + node.fixed.height)
```

Comment:
I implemented it the same way before, but then decided to extract the function, since `< 0` and `> 100` were repeating so often.

Reply:
Out of range just checks a bit too much for the last four lines :) (and I added the last two, since width and height could have been negative without noticing)